src/main/java/io/cdap/plugin/switchcase/route/BasicPortSpecification.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.switchcase.route;
+
+import com.google.common.base.Joiner;
+import io.cdap.cdap.etl.api.FailureCollector;
+
+class BasicPortSpecification extends PortSpecification {
+  private final BasicRoutingFunction routingFunction;
+  private final String parameter;
+  private final FailureCollector collector;
+
+  BasicPortSpecification(String name, RoutingSwitch.Config.FunctionType functionType, String parameter,
+                         FailureCollector collector) {
+    super(name);
+    this.routingFunction = fromFunctionType(functionType);
+    this.parameter = parameter;
+    this.collector = collector;
+  }
+
+  BasicRoutingFunction getRoutingFunction() {
```

Comment:
public ?

Reply:
same here. Only need package-private access right now. If I use it outside the package later, I'll make it public then. But can start with package-private.